packages/kotti-ui/source/props.ts

```diff
@@ -42,62 +51,95 @@ const propValidator = <SCHEMA extends z.ZodTypeAny>(
 	return validator
 }
 
-// eslint-disable-next-line sonarjs/cognitive-complexity
-const walkShape = <SHAPE extends z.ZodTypeAny>(shape: SHAPE): Set<string> => {
+/**
+ * Collects all unique types used in the top level of a Zod schema
+ */
+const walkSchemaTypes = <SCHEMA extends z.ZodTypeAny>(
+	schema: SCHEMA,
+): Set<z.ZodFirstPartyTypeKind> => {
 	/* eslint-disable no-console */
-	const types = new Set<string>()
+	const typeName = schema._def.typeName as z.ZodFirstPartyTypeKind
 
-	let shapePointer = shape
+	if (DEBUG_WALK_SHAPE)
```

Comment:
minor: u forgot to rename the debug flags to `DEBUG_WALK_SCHEMA_TYPES`

Reply:
Good catch